### ISSUE TITLE
Histogram & Time-series widget out of bounds filtering

### ISSUE THREAD

**Issue by noguerol** [CartoDB/cartodb#12480]
### Context
Based on the works underdone in the time-series milestone and more specifically in aggregations https://github.com/CartoDB/cartodb/issues/12324 and draggable filters https://github.com/CartoDB/cartodb/pull/12347 , we have to further discuss how to tackle the out-of-bounds problem that arises when filtering an histogram + a second widget both in Time-series and Histograms. 

@CartoDB/design has already promoted the visual solution: https://github.com/CartoDB/design/issues/878

Directly copied from that ticket: 

The cause of the problem is that the filters in histogram related widgets are based on the bin indexes and not the values. That is, the filter goes from bins 3 to 7 instead of saying the filter goes from 3200 to 6500. Internally, it's done like that.

That's why when we combine several filters the result gets weird. After filtering in another widget, the data domain of the other widgets change. So now the bin 3 doesn't begin at 3200 but at another value, based on the other widget filter. This way works when you only filter yourself. The funny things start when you have a filter influenced by another filter.

The implications of this behaviour are huge:

In cartodb.js the histogram-dataview-model and the histogram-data-model get this lo_index and hi_index for the filters. They have also min and max that have the real values of the filters, but are not used for visualization (only if the other ones are not present).
This two models are used by 4 different components: histogram, histogram zoom, time-series and animated time-series. Each of them have different classes for the widget UI logic. But all of them share the chart component and the two datamodels in cartodb.js. So one change in chart for one component can break the other three.
We cannot simply change lo_index and hi_index to min and max because it would break ALL existing widgets with applied filters. So, somehow, a new way to save filters (based on value) should coexist with the current one based on indexes.
Summing it up, this is not a straightforward task only related to visualization. It should ve very well thought and tested.

**Comment by rjimenezda** [comment]
I found an issue that I believe is related to this & some recent changes in how the time slider work. Can't reproduce consistently, but having a time series + other histograms and filtering & clearing filters can break the playback.

**Comment by noguerol** [comment]
Try to reproduce and report it in a new ticket, then 😘

**Comment by ivanmalagon** [comment]
Blocked until we implement sending `aggregation=auto` in https://github.com/CartoDB/cartodb/issues/12324

**Comment by xavijam** [comment]
Unblocked, aggregations free!

**Comment by ivanmalagon** [comment]
As a result of another issue acceptance, several issues related to axis labels have appeared. Since this ticket tackles the range filter, I think is suitable to add them here to be addressed.

Copypasting from those [comments](https://github.com/CartoDB/cartodb/issues/12088#issuecomment-321227542).

- [ ] Ok, so about the labels. To prevent that overlapping its better to align the label not to the dragger but to the bin itself. I attach some mocks:

![image](https://user-images.githubusercontent.com/70603/29119302-e59bec76-7d05-11e7-88ab-2266ff422828.png)


![image](https://user-images.githubusercontent.com/70603/29119306-ea4b2fca-7d05-11e7-9f52-5288b34c7f6d.png)

- [ ] Which drives us to a new bug, the cropped label on the time series limits:

![image](https://user-images.githubusercontent.com/70603/29119324-01ab5e38-7d06-11e7-8c29-d0255342d5aa.png)


- [ ] I found this bug. We shouldn't be able to place the handle in the middle of the bucket. 

![bucket](https://user-images.githubusercontent.com/16247628/29168324-b417dc72-7dce-11e7-81cf-f6f5f33443b7.gif)

- [ ] Lastly, when selecting only one bucket, we shouldn't display the value of that single bucket twice. There should only be one, above and in the middle of the bucket: 

![screen shot 2017-08-10 at 13 02 03](https://user-images.githubusercontent.com/16247628/29168377-e293b0b2-7dce-11e7-95b1-c4b73bf81f1e.png)

**Comment by xavijam** [comment]
First work done in `12480-oob-filter` branch.
